rest_api/limsdb/queries/factories.py

```diff
@@ -140,33 +140,67 @@ def project_info(session):
     project_name = match.get('project_id')
     project_status = match.get('project_status', 'open')
     all_projects = defaultdict(data_models.ProjectInfo)
-    for info in queries.get_project_info(session, project_name, udfs=['Number of Quoted Samples'],
+    for info in queries.get_project_info(session, project_name, udfs='all',
                                          project_status=project_status):
-        pjct_name, open_date, close_date, firstname, lastname, udf_name, nb_quoted_samples = info
+        pjct_name, open_date, close_date, firstname, lastname, udf_name, udf_value = info
         all_projects[pjct_name].project_id = pjct_name
         all_projects[pjct_name].open_date = open_date
         all_projects[pjct_name].close_date = close_date
         all_projects[pjct_name].researcher_name = '%s %s' % (firstname, lastname)
-        all_projects[pjct_name].nb_quoted_samples = nb_quoted_samples
+        all_projects[pjct_name].udfs[udf_name] = udf_value
 
     return [p.to_json() for p in all_projects.values()]
 
 
+def _resolve_artifact_reagent_labels(session, artifact_ids):
+    """
+    This function takes a list of artifact ids and return for each artifact the sample and reagent label pair.
+    If the artifact id correspond to a pool, it resolve multiple sample/reagent label pairs.
+    Otherwise it only returns one.
+    """
+    res = queries.artifact_reagent_labels(session, artifact_ids)
+    artifact_to_label = defaultdict(set)
+    for r in res:
+        artifact_id, ancestor_id, reagent_label, sample_name = r
+        artifact_to_label[(artifact_id, ancestor_id)].add((artifact_id, reagent_label, sample_name))
+    return set([
+        artifact_to_label[(artifact_id, ancestor_id)].pop()
+        for (artifact_id, ancestor_id) in artifact_to_label
+        if len(artifact_to_label[(artifact_id, ancestor_id)]) == 1
+    ])
+
+
 def run_status(session):
     """Query the LIMS database for run information"""
     kwargs = retrieve_args()
     time_since = kwargs.get('createddate', None)
     status = kwargs.get('status', None)
-    all_runs = defaultdict(data_models.Run)
+    match = kwargs.get('match', {})
+    run_id = match.get('run_id')
+    run_ids = match.get('run_ids')
+    if run_id:
+        run_ids = [run_id]
```

Comment:
Why is `run_id` being retrieved twice - as `run_id` and `run_ids`?

Reply:
This is to support one or multiple run_ids requested.
if run_id is provided then it will take precedence